projects/plugins/social/src/class-note.php

```diff
@@ -11,10 +11,45 @@
  * Register the Jetpack Social Note custom post type.
  */
 class Note {
+	const JETPACK_SOCIAL_NOTE_CPT                      = 'jetpack-social-note';
+	const JETPACK_SOCIAL_REWRITE_RULES_LAST_FLUSHED_AT = 'jetpack_socia_rewrite_rules_last_flushed_at';
+
+	/**
+	 * Check if the feature is enabled.
+	 */
+	public function enabled() {
+		return defined( 'JETPACK_SOCIAL_NOTES_ENABLED' ) && constant( 'JETPACK_SOCIAL_NOTES_ENABLED' );
+	}
+
+	/**
+	 * Initialize the Jetpack Social Note custom post type.
+	 */
+	public function init() {
+		flush_rewrite_rules();
+		if ( ! static::enabled() ) {
+			return;
+		}
+		static::register_cpt();
+		add_action( 'wp_insert_post_data', array( new Note(), 'set_empty_title' ), 10, 2 );
+	}
+
+	/**
+	 * Set the title to empty string.
+	 *
+	 * @param array $data The Post Data.
+	 * @param array $post The Post.
+	 */
+	public static function set_empty_title( $data, $post ) {
+		if ( self::JETPACK_SOCIAL_NOTE_CPT === $post['post_type'] && 'auto-draft' === $post['post_status'] ) {
+			$data['post_title'] = '';
+		}
+		return $data;
+	}
+
 	/**
 	 * Register the Jetpack Social Note custom post type.
 	 */
-	public static function register() {
+	public static function register_cpt() {
 		if ( ! defined( 'JETPACK_SOCIAL_NOTES_ENABLED' ) || ! constant( 'JETPACK_SOCIAL_NOTES_ENABLED' ) ) {
 			return;
 		}
```

Comment:
I think you're not going to like me! I wonder if we should swap to using an option to determine if the feature is enabled. Then we can have a function that toggles the feature and flushes the rewrite rules. The advantage being that we can hook that toggle function up to the UI on the admin page, and we'll need an option to determine if the feature is enabled anyway, so it covers both needs.

Reply:
Thanks @pablinos, that's a good idea. I have done this, but flushing the rules needs to be done after the CPT is registered and the registration can be done only on the init hook, not when the feature is enabled on/off. 

I have the changes up in https://github.com/Automattic/jetpack/pull/35462/commits/996039c3525db560f24352e8ef23e6db1ad41cd5, let me know what you think of that. I have added a test as well.